### PR TITLE
Big package support

### PR DESCRIPTION
**Pull request by thetechnick**
Todo
- [x] Implement garbage collection of unreferenced ObjectSlices
- [x] Tests ;)

### CODE REVIEW

File: apis/core/v1alpha1/objectslices_types.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// ObjectSlices are referenced by ObjectSets or ObjectDeployments and contain objects to
+// limit the size of ObjectSet and ObjectDeployments when big packages are installed.
```
Comment (garrettthomaskth) [inline apis/core/v1alpha1/objectslices_types.go:6]:
```suggestion
// limit the size of ObjectSets and ObjectDeployments when big packages are installed.
```

File: integration/package_test.go [hunk 2]
```diff
@@ -18,6 +20,7 @@ func TestPackage_success(t *testing.T) {
 		name             string
 		pkg              client.Object
 		objectDeployment client.Object
+		postCheck        func(ctx context.Context, t *testing.T)
```
Comment (garrettthomaskth) [inline integration/package_test.go:23]: `sliceCheck`?

File: docs/api-reference.md [hunk 3]
```diff
@@ -105,14 +109,16 @@ spec:
           app.kubernetes.io/name: example-operator
   lifecycleState: Active
   phases:
-  - class: sit
-    name: dolor
+  - class: amet
```
Comment (eqrx) [inline docs/api-reference.md:112]: why this change? :D

**Review by eqrx — APPROVED** [review]
lgtm but please take this with a grain of salt since I am new to PKO :D

also what @garrettthomaskth  said.
